### PR TITLE
New Serverless Pattern - s3-b2bi-s3- EDI to JSON tranformation

### PR DESCRIPTION
**Pull request by biswanathmukherjee**
*Issue #, if available:*

*Description of changes:*

This sample project demonstrates how AWS B2B Data Interchange can transform EDI Files into JSON as per the transformation logic.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: s3-b2bi-s3/template2.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+AWSTemplateFormatVersion: '2010-09-09' 
+Transform: AWS::Serverless-2016-10-31
+Description: SAM Template that creates an EC2 Instance, an Amazon S3 Express One Zone directory bucket and required IAM Role to access the bucket from the instance
```
Comment (bfreiberg) [inline s3-b2bi-s3/template2.yaml:3]: The description does not match the contents of the file

File: s3-b2bi-s3/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  "testing": {
+    "text": [
+      "See the GitHub repo for detailed testing instructions."
+    ]
+  },
+  "cleanup": {
+    "text": [
+      "Delete the Amazon S3 input bucket content: <code>aws s3 rm s3://{EDI214InputBucketName} --recursive</code> --region {my-region}",
+      "Delete the Amazon S3 input bucket content: <code>aws s3 rm s3://{EDI214OutputBucketName} --recursive</code> --region {my-region}",
```
Comment (bfreiberg) [inline s3-b2bi-s3/example-pattern.json:50]: please double check the closing code tags. Also, the second line should refer to the output bucket

File: s3-b2bi-s3/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      "File upload action triggers AWS B2B Data Interchange Transformers to transform the X12 EDI file into JSON format.",
+      "The transformed JSON file is uploaded into the output Amazon S3 bucket."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/s3-b2bi-s3",
+      "templateURL": "serverless-patterns/s3-b2bi-s3",
+      "projectFolder": "s3-b2bi-s3",
+      "templateFile": "template.yaml"
```
Comment (bfreiberg) [inline s3-b2bi-s3/example-pattern.json:20]: there is no such file in the folder

File: s3-b2bi-s3/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        "text": "AWS Blog - Manage EDI at scale with new AWS B2B Data Interchange",
+        "link": "https://aws.amazon.com/blogs/aws/introducing-aws-b2b-data-interchange-simplified-connections-with-your-trading-partners/"
+      }
+    ]
+  },
+  "deploy": {
+    "text": [
+      "See the GitHub repo for detailed deployment instructions.",
+      "sam deploy -g -t template1.yaml",
+      "sam deploy -g -t template2.yaml"
```
Comment (bfreiberg) [inline s3-b2bi-s3/example-pattern.json:39]: Why do you need two templates? Can they have more descriptive names?

File: s3-b2bi-s3/template1.yaml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Description: This SAM Template demonstrates how AWS B2B Data Interchange can transform EDI Files into JSON as per the transformation logic
+
+# Get the required input parameters
+Parameters:
+  BusinessName:
+    Type: String
+    Description: Please enter the business name
+  Email:
+    Type: String
+    Description: Please enter email id
```
Comment (bfreiberg) [inline s3-b2bi-s3/template1.yaml:12]: is it required to be a valid email pattern? If so, add properties to check and assist the user, see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties

File: s3-b2bi-s3/template1.yaml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                aws:SourceAccount:
+                  Ref: AWS::AccountId  
+
+  # Create output bucket for EDI file
+  EDI214OutputBucket: 
+    Type: AWS::S3::Bucket
+    Properties: 
+      BucketName: !Sub "edi-214-output-${AWS::AccountId}-${AWS::Region}"
+
+  # Create bucket policy to allow read operations from AWS B2B Data Interchange
```
Comment (bfreiberg) [inline s3-b2bi-s3/template1.yaml:89]: the permissions granted below are not read operations.

File: s3-b2bi-s3/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+   ```bash
+   sam deploy -g -t template2.yaml
+   ```
+8. During the prompts:
+
+   - Enter a stack name. I am assuming stack name to be `b2bi-stack2`.
+   - Enter the desired AWS Region. Please refer to the documentation for the list of supported regions.
+   - Enter `BusinessName`, `Email`, `Phone`, `EDI214InputBucketName`, `EDI214OutputBucketName`, `MyProfileId` and `MyTransformerId`. Please note the last four parameter values should be given as per the output from the `b2bi-stack1` output. 
+
+   Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
```
Comment (bfreiberg) [inline s3-b2bi-s3/README.md:51]: I don't understand how this relevant here. Could you elaborate please?

File: s3-b2bi-s3/template2.yaml (hunk 1 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/template2.yaml]: Fixed.

File: s3-b2bi-s3/example-pattern.json (hunk 3 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/example-pattern.json]: updated.

File: s3-b2bi-s3/template1.yaml (hunk 5 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/template1.yaml]: Fixed.

File: s3-b2bi-s3/template1.yaml (hunk 6 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/template1.yaml]: Fixed.

File: s3-b2bi-s3/README.md (hunk 7 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/README.md]: Updated. Please let me know if anything else to change.

File: s3-b2bi-s3/example-pattern.json (hunk 4 above)
Comment (biswanathmukherjee) [reply s3-b2bi-s3/example-pattern.json]: I have updated the names. Please let me know if they are fine now.

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution. A DA will merge your request soon

**Pull request merged**